### ISSUE TITLE
[Screen Readers-Technology Enabled Continuous Auditing-Sub Audit Template]:NVDA focus is not moving to pop up  error message upon invoking 'Submit'Button leaving the mandatory fields blank.

### ISSUE THREAD

**Issue by Amit8527510735** [microsoft/BotFramework-WebChat#3284]
![MAS3 3 1_NVDA is not reading error message](https://user-images.githubusercontent.com/53263029/85977061-7807e500-b9f9-11ea-814a-4d4eafa8680f.png)
URL: https://auditbotui-ppe.azurewebsites.net/

Image

Actual Result:
NVDA focus is not moving to pop up  error message upon invoking 'Submit'Button leaving the mandatory fields blank.
Observation: When 'Submit' button is invoked leaving the mandatory fields blank then the NVDA focus remains on the 'Submit Button' itself.

Expected Result:
NVDA focus should move  to pop up  error message upon invoking 'Submit'Button leaving the mandatory fields blank.
Ex:When the Submit is invoked NVDA should narrate as "Information, Please correct the following errors, TRID cannot be empty, TRID should be greater than zero, Purchase Order Number cannot be empty"

Note:  For a visual user, error caused fields should be highlighted with red color border indicating they need come modification/they need attention.

Other Instances:
Same issue  is observed in the below screens also:
1. SG Membership Request& SG Membership  With File Upload
Image

When the Submit is invoked by leaving the mandatory fields blank, The NVDA focus should move to first error field and NVDA should narrate as Information, Please correct the following errors, TRID cannot be empty, TRID should be greater than zero, Please enter at least one security group names that you want to enlist.
Note:  For a visual user, error caused fields should be highlighted with red color border indicating they need come modification/they need attention.

MAS Reference:
MAS 3.3.1 - Error Identification

**Comment by cwhitten** [comment]
@Amit8527510735 I've transferred this to the appropriate project. Can you reapply the right labels?

**Comment by Amit8527510735** [comment]
@cwhitten Thanks for transferring bug to correct project. I have added labels, can you please assign bug to correct owner as this bugs still not assigned to anyone.

**Comment by scheyal** [comment]
@cwhitten  - can you please propose an owner & re-assign?

**Comment by compulim** [comment]
From the screenshot, the error bubble itself is not clickable and not interactive. Thus, the focus must not be moved there.

We provided a live region for screen reader user to get notifications on incoming messages. When the error bubble shows up, the screen reader will read its content to the user.

In 3.3.1, https://www.w3.org/WAI/WCAG21/Understanding/error-identification.html, under Techniques > Sufficient Techniques > Siutation B > point number 2. Live Regions is consider a sufficient technique for success criteria of 3.3.1 (https://www.w3.org/WAI/WCAG21/Techniques/aria/ARIA19).

Can you list out reasons why ARIA19 is not a sufficient technique for 3.3.1?

**Comment by Amit8527510735** [comment]
@compulim 
We have discussed this with our SME, he provided feedback. Please find details below:
"The error/message should be read as soon as it appears on the screen. For accomplishing this, please advise the development to add role=”alert” to the error fragment which gets added in the dom dynamically." 
If any query please let me know.

**Comment by compulim** [comment]
Hi @Amit8527510735, can you upgrade the Web Chat from v3 to v4.9.2 and see if it repro?

v3 is being deprecated and is on life support now.

**Comment by daveta** [comment]
Hi @Amit8527510735 : Please reactivate if you have any problems with the suggestion..
Thanks, -Dave

**Comment by Amit8527510735** [comment]
#FirstParty:CSEO; CSEO:WebChat;

**Comment by Amit8527510735** [comment]
#HCL;#Accessibility;#HCL_BotFramework_WebChat;A11yMAS;WCAG3.3.1
